main.go

```diff
@@ -272,6 +272,18 @@ func enterProvider(wg *sync.WaitGroup, p provider.Provider, mergerCH <-chan []st
 			}
 
 			var err error
+			createNotify := func(err error, t time.Duration) {
+				log.Errorf("%v at time: %v", err, t)
+				if awsErr, ok := err.(awserr.Error); ok {
```

Comment:
1. Done. 
2. created custom errors in provider/error.go that uses provider package. 
However I had to move newProvider to main.go to avoid cyclic dependencies not sure if there an easier way of doing this sort of shared errors between providers.